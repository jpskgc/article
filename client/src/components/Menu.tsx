import React from 'react';
import {Container, Menu} from 'semantic-ui-react';
import firebase from '../firebase/firebase';
import {Link} from 'react-router-dom';

interface UserStatus {
  user: firebase.User | null;
}

class FixedMenuLayout extends React.Component<{}, {}> {
  public state: UserStatus = {
    user: null,
  };

  componentDidMount() {
    firebase.auth().onAuthStateChanged(user => {
      this.setState({user});
    });
  }

  logout() {
    firebase.auth().signOut();
  }

  render() {
    return (
      <div>
        <Menu fixed="top" inverted>
          <Container>
            <Menu.Item as="a" header>
              Project Name
            </Menu.Item>
            <Menu.Item as="a" href="/">
              Home
            </Menu.Item>
            {this.state.user ? (
              // <Menu.Item as="a" href="/post">
              //   Post
              // </Menu.Item>
              <Menu.Item name="post">
                <Link to="/post">Post</Link>
              </Menu.Item>
            ) : (
              <div />
            )}
            {this.state.user ? (
              <Menu.Menu position="right">
                <Menu.Item name="setting">
                  {/* TODO 選択範囲 */}
                  <Link to="/setting">Setting</Link>
                </Menu.Item>
                <Menu.Item name="logout" onClick={this.logout}>
                  <Link to="">Log-out</Link>
                </Menu.Item>
              </Menu.Menu>
            ) : (
              <Menu.Menu position="right">
                <Menu.Item name="login" href="/login">
                  Log-in
                </Menu.Item>
              </Menu.Menu>
            )}
          </Container>
        </Menu>
      </div>
    );
  }
}

export default FixedMenuLayout;
