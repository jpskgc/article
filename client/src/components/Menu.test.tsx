import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import Menu from './Menu';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Menu />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
