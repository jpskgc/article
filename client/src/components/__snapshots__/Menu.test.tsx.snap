// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="ui inverted top fixed menu"
  >
    <div
      className="ui container"
    >
      <a
        className="header item"
        onClick={[Function]}
      >
        Project Name
      </a>
      <div
        className="item"
        onClick={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
      <div />
      <div
        className="right menu"
      >
        <div
          className="item"
          onClick={[Function]}
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Log-in
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
